Documentation/sources/push-notification/manual-handling-and-tracking.md

```diff
@@ -0,0 +1,149 @@
+# Manual display and tracking of push notification
+
+## Overview
+This document provides instructions on understanding notification from Adobe Journey Optimizer and subsequently building, displaying, and monitoring push notifications manually with the AEPMessaging extension.
+
+## Sync the push token
+
+To register the push token with Adobe Experience Platform, add the following code to the Application class's `onCreate()` method:
+
+```java
+public class YourApp extends Application {
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        FirebaseMessaging.getInstance().getToken().addOnCompleteListener(new OnCompleteListener<String>() {
+
+            @Override
+            public void onComplete(@NonNull Task<String> task) {
+                if (task.isSuccessful()) {
+                    String token = task.getResult();
+                    MobileCore.setPushIdentifier(token);
+                }       
+            }
+        });
+    }
+}
+```
+
+## Buidling and displaying notification
+
+1. In `onMessageReceived` method of `YourAppFirebaseMessagingService` class, create a `MessagingPushPayload` object from the remoteMessage. `MessagingPushPayload` will unpack the remoteMessage and provide APIs for getting attributes used for creating the push notification.
```

Comment:
how about: "..., create a `MessagingPushPayload` object from the received remoteMessage object".